### PR TITLE
Feature/persistence rocksdb 2 btc 500

### PR DESCRIPTION
**Pull request by blackyblack**

### CODE REVIEW

File: src/blockchain/alt_block_tree.cpp [hunk 1]
```diff
@@ -243,19 +240,19 @@ void AltTree::payloadsToCommands(const VbkBlock& block,
...
+  saveBlocksAndTip(storage, *this);
   return state.IsValid();
 }
 
-bool AltTree::loadFromStorage(const PopStorage& storage,
+bool AltTree::loadFromStorage(PopStorage& storage,
                               ValidationState& state) {
-  bool ret = loadBlocks(storage, vbk().btc(), state);
+  bool ret = loadAndApplyBlocks(storage, vbk().btc(), state);
   if (!ret) return state.IsValid();
```
Comment (Warchant) [inline src/blockchain/alt_block_tree.cpp:252]: should probably be 
```
if(!loadAndApplyBlocks(...)) {
  return false;
}
```

File: test/e2e/scenario_0/scenario_0.cpp [hunk 2]
```diff
@@ -60,7 +60,13 @@ struct Scenario0 : public ::testing::Test {
...
   std::shared_ptr<Config> config;
   std::shared_ptr<AltTree> alt;
-  PayloadsStorage storage{};
+  std::shared_ptr<PayloadsRepository<ATV>> prepoAtv =
+      std::make_shared<PayloadsRepositoryInmem<ATV>>();
+  std::shared_ptr<PayloadsRepository<VTB>> prepoVtb =
+      std::make_shared<PayloadsRepositoryInmem<VTB>>();
+  std::shared_ptr<PayloadsRepository<VbkBlock>> prepoBlocks =
+      std::make_shared<PayloadsRepositoryInmem<VbkBlock>>();
+  PayloadsStorage storage{prepoAtv, prepoVtb, prepoBlocks};
```
Comment (Warchant) [inline test/e2e/scenario_0/scenario_0.cpp:69]: can you pls create factory function that creates PayloadsStorage and use it here

File: include/veriblock/storage/block_repository_rocks.hpp [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+    write_options.disableWAL = true;
+    rocksdb::Status s = _db->Delete(write_options, _columnHandle, makeRocksSlice(hash));
+    if (!s.ok() && !s.IsNotFound()) {
+      throw db::DbError(s.ToString());
+    }
+    return true;
+  }
+
+  void clear() override {
+    // call BlockRepositoryRocksManager.clear() instead
```
[the remark this answers is not in the document]
Comment (Warchant) [reply include/veriblock/storage/block_repository_rocks.hpp:316]: if not, maybe use `std::terimnate` or `VBK_ASSERT(false && "msg..")`

File: include/veriblock/storage/pop_storage.hpp [hunk 4]
```diff
@@ -3,48 +3,70 @@
...
   virtual ~PopStorage() = default;
-  PopStorage() {}
+  PopStorage(std::shared_ptr<BlockRepository<BlockIndex<BtcBlock>>> brepoBtc,
+             std::shared_ptr<BlockRepository<BlockIndex<VbkBlock>>> brepoVbk,
+             std::shared_ptr<BlockRepository<BlockIndex<AltBlock>>> brepoAlt,
+             std::shared_ptr<TipsRepository<BlockIndex<BtcBlock>>> trepoBtc,
+             std::shared_ptr<TipsRepository<BlockIndex<VbkBlock>>> trepoVbk,
+             std::shared_ptr<TipsRepository<BlockIndex<AltBlock>>> trepoAlt,
+             std::shared_ptr<PayloadsRepository<VbkEndorsement>> erepoVbk,
+             std::shared_ptr<PayloadsRepository<AltEndorsement>> erepoAlt)
```
[the remark this answers is not in the document]
Comment (blackyblack) [reply include/veriblock/storage/pop_storage.hpp:23]: I think it is better to have them separated since they have quite different data structures.

**Pull request merged**